scripts/run-ci-e2e-tests.js

```diff
@@ -16,173 +16,236 @@
  * --android - to test only android application end to end
  * --js - to test that JS in the application is compilable
  * --skip-cli-install - to skip react-native-cli global installation (for local debugging)
+ * --retries [num] - how many times to retry possible flaky commands: npm install and running tests, default 1
  */
 /*eslint-disable no-undef */
 require('shelljs/global');
-var spawn = require('child_process').spawn;
+
+const spawn = require('child_process').spawn;
+const argv = require('yargs').argv;
 const path = require('path');
 
 const SCRIPTS = __dirname;
 const ROOT = path.normalize(path.join(__dirname, '..'));
 
-const TEMP=exec('mktemp -d /tmp/react-native-XXXXXXXX').stdout.trim();
+const TEMP = exec('mktemp -d /tmp/react-native-XXXXXXXX').stdout.trim();
 // To make sure we actually installed the local version
 // of react-native, we will create a temp file inside the template
 // and check that it exists after `react-native init
 const MARKER_IOS = exec(`mktemp ${ROOT}/local-cli/generator-ios/templates/app/XXXXXXXX`).stdout.trim();
 const MARKER_ANDROID = exec(`mktemp ${ROOT}/local-cli/generator-android/templates/src/XXXXXXXX`).stdout.trim();
-
+const numberOfRetries = argv.retries || 1;
 let SERVER_PID;
 let APPIUM_PID;
+let exitCode;
 
-const args = process.argv.slice(2);
-
-function cleanup(errorCode) {
-  if (errorCode !== 0) {
-    cat(`${TEMP}/server.log`);
-    cat(`/usr/local/Cellar/watchman/3.1/var/run/watchman/${process.env.USER}.log`);
-  }
-  rm(MARKER_IOS);
-  rm(MARKER_ANDROID);
 
-  if(SERVER_PID) {
-    echo(`Killing packager ${SERVER_PID}`);
-    exec(`kill -9 ${SERVER_PID}`);
-    // this is quite drastic but packager starts a daemon that we can't kill by killing the parent process
-    // it will be fixed in April (quote David Aurelio), so until then we will kill the zombie by the port number
-    exec("lsof -i tcp:8081 | awk 'NR!=1 {print $2}' | xargs kill");
-  }
-  if(APPIUM_PID) {
-    echo(`Killing appium ${APPIUM_PID}`);
-    exec(`kill -9 ${APPIUM_PID}`);
+/**
+ * Try executing a function n times recursively.
+ * Return 0 the first time it succeeds
+ * Return code of the last failed commands if not more retries left
+ * @funcToRetry - function that gets retried
+ * @retriesLeft - number of retries to execute funcToRetry
+ * @onEveryError - func to execute if funcToRetry returns non 0 
+ */
+function tryExecNTimes(funcToRetry, retriesLeft, onEveryError) {
+  const code = funcToRetry();
```

Comment:
nit: Maybe `exitCode`? For a while I thought `code` was a function, not sure why..